File: findDuplicates/findDuplicates.go
/*
Same prompt and solution as javascript version
*/
package main

import "math"

func findDuplicates(numbers []int) []int {
	result := make([]int, 0)
	for i := range numbers {
		position := int(math.Abs(float64(numbers[i])) - 1)
		if numbers[position] < 0 {
			result = append(result, position+1)
		} else {
			numbers[position] = -numbers[position]
		}
	}
	return result
}

File: Leetcode/findDuplicates/findDuplicates.js
const findDuplicates = nums => {
  let mem = [];
  let result = [];
  for (let i = 0; i < nums.length; i++) {
    if (mem[nums[i]]) {
      result.push(nums[i])
    } else {
      mem[nums[i]] = true
    }
  }
  return result
}

module.exports = findDuplicates